infra/enrolment-records-store/src/main/java/com/simprints/infra/enrolment/records/store/EnrolmentRecordRepositoryImpl.kt

```diff
@@ -109,4 +115,30 @@ internal class EnrolmentRecordRepositoryImpl(
             project = project
         )
     }
+
+    override suspend fun count(query: SubjectQuery, dataSource: BiometricDataSource): Int =
+        when (dataSource) {
```

Comment:
I would suggest something like:
```
fun fromIdentitySource(dataSource: BiometricDataSource): IdentityDataSource = 
    when (dataSource) {
       BiometricDataSource.SIMPRINTS -> localDataSource
       BiometricDataSource.COMMCARE -> commCareDataSource
    }
```

And then on callers side could be simplified to 
```
enrolmentRecordRepo.fromIdentitySource(dataSource).<method>()
```

Reply:
It would simplify the repo implementation a little but would expose the IdentityDataSource directly to callers. Currently we simply forward the calls but having it pass through the repo reserves the right to alter the logic in the future. So not sure it's worth it.

Reply:
IMO, such DSL-like API could be better, but that is a matter of taste. Although, you can still use private a method for matching internally to reduce duplication a bit.

Reply:
Also, `IdentityDataSource` is currently public, so it is already exposed :)